TweetyBird/twitter_utils.py

```diff
@@ -0,0 +1,152 @@
+import re
+
+from replit import db
+import tweepy
+
+from .discord_utils import send_discord_message
+from .settings import Twitter_API_PK, Twitter_API_SK, Twitter_Access_Token, Twitter_Access_Secret
+
+
+# Modify the default Twitter Stream, and overwrite the default on_status call
+class TwitterStreamListener(tweepy.StreamListener):
+    def on_status(self, status):
+        send_discord_message(format_tweet(status))
+
+
+# Authenticate with Twitter
+auth = tweepy.OAuthHandler(Twitter_API_PK, Twitter_API_SK)
+auth.set_access_token(Twitter_Access_Token, Twitter_Access_Secret)
+api = tweepy.API(auth, wait_on_rate_limit=True, wait_on_rate_limit_notify=True)
+try:
+    api.verify_credentials()
+    print("Authentication OK")
+except Exception:
+    print("Error during authentication")
+
+# Tweepy Stream setup
+myStreamListener = TwitterStreamListener()
+myStream = tweepy.Stream(auth=api.auth, listener=myStreamListener)
+
+# Filter requires list of ID's and a filter level (none, low, medium, high)
+myStream.filter(follow=db["TwitterFollows"], is_async=True, filter_level="medium")
+
+#Regex Cached URL Finder
+p = re.compile('(https://t.co/[a-zA-Z0-9]{10})')
+
+
+def update_following(twitter_user):
+    """
+    If the provided user is a valid Twitter user:
+      - Add the user to the following
+      - Init db if this is the first user
+    """
+    msg = "Not a valid Twitter Account"
+    if is_valid_twitter_user(twitter_user):
+        twitter_user_id_string = api.get_user(twitter_user).id_str
+        # Ensure database key exists
```

Comment:
I'm not familiar with replit. But if this is just a normal python dictionary you could use this and not worry about having to create the database key since it'll give you an empty list if that key doesn't exist.

```
following = db.get("TwitterFollows", [])
following.append(twitter_user_id_string)
```

https://coderscat.com/python-dict-setdefault-and-getdefault/